### PR TITLE
[TASK] Rewrite AJAX chapter

### PR DESCRIPTION
**Pull request by andreasfernandez**
The chapter about AJAX requests is rewritten to explain how the AJAX API coming with TYPO3 v10.3 works and how a controller should be written to accept and handle such requests.

### CODE REVIEW

File: Documentation/ApiOverview/Ajax/Backend/Index.rst [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+   A reason for the given status code. If omitted, the default for the used status code will be used.
+
+.. code-block:: php
+
+   public function doSomethingAction(ServerRequestInterface $request): Response
+   {
+       // our previous computation
+
+       return new Response(json_encode(['result' => $result], 200, ['Content-Type' => 'application/json; charset=utf-8']))
```
Comment (simonschaufi) [inline Documentation/ApiOverview/Ajax/Backend/Index.rst]: The class `JsonResponse` would fit better here.

Comment (andreasfernandez) [reply Documentation/ApiOverview/Ajax/Backend/Index.rst]: JsonResponse is internal API, tgat's why I didn't use it here.

Comment (simonschaufi) [reply Documentation/ApiOverview/Ajax/Backend/Index.rst]: Mmh ok even though it's much cleaner.

File: Documentation/ApiOverview/Ajax/Index.rst [hunk 2]
```diff
@@ -0,0 +1,14 @@
+.. include:: ../../Includes.txt
+
+.. _ajax:
+
+====
+AJAX
+====
+
```
Comment (DanielSiepmann) [inline Documentation/ApiOverview/Ajax/Index.rst]: Shouldn't we add an info that this is only for Ajax within TYPO3 Backend?

File: Documentation/ApiOverview/Ajax/Backend/Index.rst [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+.. _ajax-backend:
+
+=======
+Backend
+=======
+
+An AJAX endpoint in the TYPO3 backend is usually implemented as a method in a regular controller. The method receives a
+request object implementing the :php:`Psr\Http\Message\ServerRequestInterface`, which allows to access all aspects of
+the requests and returns an appropriate response in a normalized way. This approach is standardized as `PSR-7`_.
```
Comment (DanielSiepmann) [inline Documentation/ApiOverview/Ajax/Backend/Index.rst]: Shouldn't it be `the request` instead of `the requests`?

**Comment by DanielSiepmann** [comment]
Everything is fine enough to be merged. Let's get the new docs up to date as soon as possible.

Still I've added some smaller comments which we might improve with another PR.

**Pull request merged**